### ISSUE TITLE
2 and more domains on the same Synapse server and/or aliases

### ISSUE THREAD

**Issue by 000xxx** [spantaleev/matrix-docker-ansible-deploy#2620]
**Is your feature request related to a problem? Please describe.**
The ability to host two or more separate domains on the same server.

**Describe the solution you'd like**
Currently it's possible to define one domain and setup server for this domain. For example, matrix.example.com . 
Idea is to host 2 independent domains on 1 matrix server. For example, to host additional domain matrix.foobar.com 

2nd idea: Also it would be great also to have opportunity to have alias matrix.example.org for current matrix.example.com with the same users.

**Comment by samip5** [comment]
Do you mean support for more synapse servers on one actual ansible host? This FR is closest I found about that.

**Comment by sgtbradley** [comment]
I do not think it is supported as of now to have multiple domains properly associated with one matrix server. You could probably have one domain redirect to the one the matrix server is on but you may have troubles trying to use that in clients and stuff. Probably a similar limitation to the fact you also can not change the domain a matrix server is running on without starting fresh

**Comment by spantaleev** [comment]
Due to us switching to Traefik, this is becoming more and more feasible.

Still, various services do not have their systemd service name and container name configurable yet (using `*_identifier` variables), so service names will be in conflict with one another.

Feel free to send PRs to make roles have a `matrix_ROLE_identifier` variable and obey it for naming their systemd services and containers!

Later on, all these `_identifier` variables can be hooked to some global variable, so they can be changed easily from a single place. The [mash-playbook](https://github.com/mother-of-all-self-hosting/mash-playbook) already supports doing this thanks to various such variables. See more details [here](https://github.com/mother-of-all-self-hosting/mash-playbook/blob/73bc7a7e5edf3e28fbdce0199d160f09f53a9e68/roles/mash/playbook_base/defaults/main.yml#L3-L32) and in its group vars file ([example](https://github.com/mother-of-all-self-hosting/mash-playbook/blob/73bc7a7e5edf3e28fbdce0199d160f09f53a9e68/group_vars/mash_servers#L167-L171)).

This Matrix playbook is still rather far from this and will need a lot of work to get there, but.. at least thanks to using Traefik, it's become a possibility.

**Comment by altsalt** [comment]
Related this upstream Synapse issue:  
https://github.com/matrix-org/synapse/issues/8885

**Comment by irgolic** [comment]
This would be great, even if it means duplicating the database etc.

**Comment by spantaleev** [comment]
Closing this as part of a backlog cleanup. Hosting multiple independent Matrix domains on one server via this playbook still requires making every role's systemd service and container names configurable (the *_identifier work mentioned earlier), and in the years since, no contributor has picked this up: most roles still hardcode their names. The second idea (domain aliases pointing to the same users) depends on Synapse gaining multi-domain support, which never landed upstream (matrix-org/synapse#8885), so it is not something this playbook can provide on its own. I am still open to pull requests that introduce identifier variables role by role, and this can be revived if a contributor appears or upstream support lands.
